pandas/core/internals/construction.py

```diff
@@ -547,19 +549,29 @@ def _ensure_2d(values: np.ndarray) -> np.ndarray:
     return values
 
 
-def _homogenize(data, index: Index, dtype: DtypeObj | None) -> list[ArrayLike]:
+def _homogenize(
+    data, index: Index, dtype: DtypeObj | None
+) -> tuple[list[ArrayLike], list[Any]]:
     oindex = None
     homogenized = []
+    # if the original array-like in `data` is a Series, keep track of this Series' refs
+    refs: list[Any] = []
 
     for val in data:
         if isinstance(val, ABCSeries):
+            orig = val
             if dtype is not None:
                 val = val.astype(dtype, copy=False)
             if val.index is not index:
                 # Forces alignment. No need to copy data since we
                 # are putting it into an ndarray later
                 val = val.reindex(index, copy=False)
-
+            if val is orig:
```

Comment:
Is this possible with CoW enabled? I think astype and reindex should create lazy copies now when no copy was made?

Reply:
Yes, astype/reindex will already give a new object, but in case of neither of those have run, we still need to take a shallow copy of the input array.

Strictly speaking, we don't need the shallow copy always; we only need the `refs` if it has one, and otherwise create new refs. But that seemed simpler with a shallow copy. If the Series already had refs, that will create an extra intermediate Block with refs, but that should directly go out of scope again.

Reply:
Yikes, yes I missed that. Was just thinking about these two not about that neither runs. Can you add a comment?

Reply:
Hmm, actually, when I create a shallow copy here, the _original_ series isn't updated to track this BlockValuesRef, should I do that?

Reply:
> Hmm, actually, when I create a shallow copy here, the _original_ series isn't updated to track this BlockValuesRef, should I do that?

Ah, no, we always create a BlockValuesRef for a Block that at a minimum refs to its own

Reply:
So, that also means I can just grab the `refs` of the Series, I suppose

Reply:
This confuses me a bit. Do you mean that the initial series references only itself? It should share the reference object with the DataFrame column and itself, e.g. ``referenced_blocks`` should have 2 elements. Is this different here?

Reply:
Wait a bit, this is still failing in the tests I added. Not sure how I missed that locally, as that's specifically the tests that I _do_ run locally ;)

Reply:
>  Do you mean that the initial series references only itself?

Yes, but I meant that this does that _before_ the dataframe is constructed (so inside `_homogenize`)

> Wait a bit, this is still failing in the tests I added

OK, just a BlockManager -> SingleBlockManager in my last "quick fixup" commit

> So, that also means I can just grab the refs of the Series, I suppose

And so on the original question of "is the shallow copy needed": no, I can of course just grab the refs of the original Series, no need for an intermediate shallow copied block.

Reply:
Ah ok, got it. 

If we guarantee to make a shallow copy later on, then the refs are sufficient yes.

Reply:
We pass the `refs` to the Block constructor, which is essentially doing a shallow copy, right?